src/components/media_manager/src/media_manager_impl.cc

```diff
@@ -298,13 +298,29 @@ void MediaManagerImpl::StopStreaming(
     int32_t application_key, protocol_handler::ServiceType service_type) {
   SDL_LOG_AUTO_TRACE();
 
-  stream_data_size_ = 0ull;
-
   if (streamer_[service_type]) {
     streamer_[service_type]->StopActivity(application_key);
   }
 }
 
+void MediaManagerImpl::SuspendStreaming(
+    int32_t application_key, protocol_handler::ServiceType service_type) {
+  SDL_LOG_AUTO_TRACE();
+  using namespace application_manager;
+  using namespace protocol_handler;
+
+  const auto app = application_manager_.application(application_key);
+  if (app) {
+    // Reset playback time and stream size when audio file has been played
+    if (ServiceType::kAudio == service_type &&
+        "socket" == settings().audio_server_type()) {
+      stream_data_size_ = 0ull;
+      socket_audio_stream_start_time_ =
```

Comment:
when streaming is suspended, i thought there could still be unstreamed data on the core side. why would you zero `stream_data_size_` here? also since you are suspending the stream, i don't see how it makes sense to set `socket_audio_stream_start_time_` here.

perhaps i am misunderstanding, when is this method `MediaManagerImpl::SuspendStreaming` called?

Reply:
@iCollin `stream_data_size_` should be set to zero here in order to properly calculate a new timeout for SuspendStreaming. This is required only for audio streaming and only for socket mode previously done in #3064 
Without that change, we had a failure of ATF script which verifies that timeout was calculated properly. As I remember there was a case when there was streaming few seconds, then 10 seconds wait and then proceed streaming. Because streaming was suspended and them resumed again, `socket_audio_stream_start_time_` was not refreshed properly and timeout time was wrong.
Actually, it is a code taken from `StopStreaming()` before my changes. `socket_audio_stream_start_time_` is just set to zero for completeness, however this is not necessary.

Reply:
@iCollin you are right, there might be an unstreamed data on the core side when stream suspension is happening. However, `stream_data_size_` is required only for calculation of the next streaming timeout so it is safe to reset it when the actual timeout has happened.